Bottom: 788f2cf8071219ba2369f7b050aabda12a7f6446
Top:    0cc82440e58392dbf95643bdffd9f9f1d5be1364
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-02-18 18:04:51 +0000

Fix a few random places where we fail badly if the board outline polygon computes to NULL

Usually, the board outline polygon should default to being the world region, but if a
polygon computation fails, it may return NULL, and we can crash in a few places if that
happens.


---

diff --git a/src/hid/common/object3d.c b/src/hid/common/object3d.c
index 405f9ab..4553186 100644
--- a/src/hid/common/object3d.c
+++ b/src/hid/common/object3d.c
@@ -251,6 +251,9 @@ object3d_from_board_outline (void)
 
   board_outline = board_outline_poly (true);
 
+  if (board_outline == NULL)
+    return NULL;
+
   /* Loop over all board outline pieces */
   pa = board_outline;
   do {
diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index 5ce8ea1..7e87a10 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -1692,6 +1692,8 @@ fill_board_outline_holes (hidGC gc, const BoxType *drawn_area)
   struct outline_info info;
 
   ensure_board_outline ();
+  if (PCB->Data->outline == NULL)
+    return;
 
   memset (&polygon, 0, sizeof (polygon));
   polygon.Clipped = PCB->Data->outline;
